### PR TITLE
Enable key-rotation ability for piped key

### LINKED ISSUE #1102: Support key rotation for piped
**What would you like to be added**:

Currently, a piped can have only one key and its key can be recreated from the web page.
At the API, the piped key is cached in memory for a time.
https://github.com/pipe-cd/pipe/blob/master/pkg/app/api/pipedtokenverifier/verifier.go#L42
This causes an "Unauthenticated" error for the piped which is using the newly created key while the cache data was not refreshed.

To deal with this problem we have some options:
- reducing the cache time (simple but not resolving the problem 100%)
- removing the cache data when the piped key was recreated (not easy)
- allowing the piped can have multiple keys (require SPEC changes)
- retrieving the key data in the datastore when it was not found in the cache or not matching with the cache data. This increases the requests to datastore for the bad key, so maybe we should cache the bad key too.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #1102

**Does this PR introduce a user-facing change?**:

```release-note
Enable key-rotation ability for piped key
```

### CODE REVIEW

**Comment by nakabonne** [comment]
"key-rotation ability" may be more appropriate for the commit message.

File: pkg/model/piped.go [hunk 1]
```diff
@@ -54,9 +56,51 @@ func GeneratePipedKey() (key, hash string, err error) {
...
+	for _, k := range p.Keys {
+		err = bcrypt.CompareHashAndPassword([]byte(k.Hash), []byte(key))
+		if err == nil {
+			return nil
+		}
+	}
+	return
+}
+
+// AddKey add a new key the list.
```
Comment (nakabonne) [inline pkg/model/piped.go:79]: super nit:
```suggestion
// AddKey adds a new key to the list.
```

**Review by nakabonne — COMMENTED** [review]
LTGM except for a super super nit comment.

**Comment by nghialv** [comment]
Thank you.
/approve

**Pull request merged**